src/classes/AN_AutoNumberService.cls

```diff
@@ -0,0 +1,385 @@
+* increment based on the last used number for the active Display Format, rather than
+* using a global number sequence for all new records for the object.
+*/
+public class AN_AutoNumberService {
+    SObjectType sObjType;
+    List<SObject> records;
+    public class AutoNumberException extends Exception {}
+
+    /**
+     * @description This constructor is currently used by the settings UI
+     * to create and activate/deactivate Auto Numbers for the passed in
+     * SObjectType.
+     */
+    public AN_AutoNumberService(SObjectType sObjType) {
+        this.sObjType = sObjType;
+    }
+
+    /**
+     * @description This constructor is currently used by the Trigger Handler
+     * framework when applying Auto Numbers to new records.
+     */
+    public AN_AutoNumberService(List<SObject> records) {
+        this.records = records;
+        this.sObjType = records.isEmpty() ? null : records[0].getSObjectType();
+    }
+
+    public void activate(String autoNumberId) {
+        List<AutoNumber__c> autoNumbersToUpdate = new List<AutoNumber__c>();
+        for (AutoNumber__c autoNumber : activeAutoNumbers) {
+            autoNumber.IsActive__c = false;
+            autoNumbersToUpdate.add(autoNumber);
+        }
+
+        autoNumbersToUpdate.add(new AutoNumber__c(Id = autoNumberId, IsActive__c = true));
+        update autoNumbersToUpdate;
+
+        if (!isTriggerHandlerEnabled) {
+            insert triggerHandler;
+        }
+    }
+
+    public void deactivate(String autoNumberId) {
+        AutoNumber__c a = new AutoNumber__c(Id = autoNumberId, IsActive__c = false);
+        update a;
+    }
+
+    public AutoNumber__c save(String autoNumberJSON) {
+        AutoNumber an = new AutoNumber(autoNumberJSON, this);
+        return an.save();
+    }
+
+    /**
+     * @description Primary method used by callers to apply Auto Numbers to the
+     * records in the trigger context that have active Auto Numbers.
+     */
+    public void apply() {
+        AutoNumber an = getAutoNumber();
+        if (an == null) {
+            return;
+        }
+
+        List<SObject> sObjectsToUpdate = new List<SObject>();
+        Integer nextAutoNumberInSequence = an.nextAutoNumber;
+        for (SObject record : records) {
+            SObject clone = record.clone(true);
+            clone.put(an.sObjField, an.getFormattedNumber(nextAutoNumberInSequence));
+            sObjectsToUpdate.add(clone);
+            nextAutoNumberInSequence++;
+        }
+
+        update sObjectsToUpdate;
+
+        // Update the AutoNumber record AFTER Successful Save
+        an.setMaxUsedNumber(nextAutoNumberInSequence - 1);
+    }
+
+    private List<AutoNumber__c> activeAutoNumbers {
+        get {
+            return [
+                    SELECT IsActive__c
+                    FROM AutoNumber__c
+                    WHERE Object_API_Name__c = :String.valueOf(sObjType)
+                    AND IsActive__c = true
```

Comment:
Totally optional, and maybe this is out of line with the letter of our apex standard, but I do like putting the "AND/OR" bits of the WHERE clause one additional tab to the right